miner/worker.go

```diff
@@ -666,20 +666,19 @@ func (w *worker) resultLoop() {
 				w.recentMinedBlocks.Add(block.NumberU64(), []common.Hash{block.ParentHash()})
 			}
 
-			// Broadcast the block and announce chain insertion event
-			w.mux.Post(core.NewMinedBlockEvent{Block: block})
-
 			// Commit block and state to database.
 			task.state.SetExpectedStateRoot(block.Root())
 			start := time.Now()
-			_, err := w.chain.WriteBlockAndSetHead(block, receipts, logs, task.state, true)
-			if err != nil {
-				log.Error("Failed writing block to chain", "err", err)
+			status, err := w.chain.WriteBlockAndSetHead(block, receipts, logs, task.state, true)
+			if status != core.CanonStatTy {
```

Comment:
when status==SideStatTy and err==nil
Log.Error will output,  but no error happens indeed

Reply:
case 1: NonStatTy & err !=nil, ==>  likely the local DB is in trouble, better not broadcast
case 2: CanonStatTy    ==>   reorg done,  should not broadcast?
case 3: SideStatTy   ==> valid block is mined, should broad?

Reply:
there are many cases which could lead to case 1:
- chain stopped
- local DB error: fail to get TD, fail to commit, invalid merkle root
- reorg failed,

Reply:
I think it is ok, but could you add some comments?

Reply:
my concern here is not output Error Log when not necessary
if err！=nil then  out put error log
if status == core.CanonStatTy then broadcast block 

but now when status!=core.CanonStatTy and err==nil, then output Error Log

Reply:
updated